odoo/tools/misc.py

```diff
@@ -1791,9 +1900,10 @@ def has_list_types(values, types):
     """
     return (
         isinstance(values, (list, tuple)) and len(values) == len(types)
-        and all(isinstance(item, type_) for item, type_ in zip(values, types))
+        and all(itertools.starmap(isinstance, zip(values, types)))
```

Comment:
Suggested by ruff, better performance.

Reply:
Testing on my side, I realized that it is indeed more efficient that way.  IIRC iterator stuff used to be slower than plain stupid code.  I never used `starmap`, because pure functional style uses more brain power for reading the code :smile:  Let it be that way, then.